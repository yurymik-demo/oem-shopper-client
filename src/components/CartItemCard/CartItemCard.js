import React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import ItemRating from "../../components/ItemRating/ItemRating";

import "./CartItemCard.scss";

export class CartItemCard extends React.Component {
  render() {
    const { item, quantity } = this.props.cartItem;
    return (
      <div className="row cart-item-card">
        <div className="col s12 m12">
          <div className="card white ">
            <div className="card-content ">
              <span className="card-title">{item.title}</span>
              <div className="right">{item.price}</div>
              <p>{item.description}</p>
              <div className="right">QTY: {quantity}</div>
            </div>
            <div className="card-action">
              <ItemRating className="left" rating={item.rating} />
              <button className="waves-effect waves-light btn btn-small red right">
                X
              </button>
              <button className="waves-effect waves-light btn btn-small orange right">
                Update
              </button>
              <Link className="right" to={`/items/${item.id}`}>
                View Details
              </Link>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default connect()(CartItemCard);
