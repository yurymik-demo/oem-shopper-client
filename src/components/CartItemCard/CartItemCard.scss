.cart-item-card {
  .card-action {
    min-height: 60px;
    .btn {
      margin-left: 8px;
    }
  }
}
