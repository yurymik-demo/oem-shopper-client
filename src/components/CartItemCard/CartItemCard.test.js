import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import { CartItemCard } from "./CartItemCard";

it("renders without crashing", () => {
  const mockItem = {
    item: {
      title: "title",
      description: "item descriotion",
      price: "123.00",
      rating: 4
    },
    quantity: 3
  };
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <CartItemCard cartItem={mockItem} />
    </BrowserRouter>,
    div
  );
  expect(div).toMatchSnapshot();
  ReactDOM.unmountComponentAtNode(div);
});
