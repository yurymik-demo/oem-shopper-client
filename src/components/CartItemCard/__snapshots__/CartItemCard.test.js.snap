// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    class="row cart-item-card"
  >
    <div
      class="col s12 m12"
    >
      <div
        class="card white "
      >
        <div
          class="card-content "
        >
          <span
            class="card-title"
          >
            title
          </span>
          <div
            class="right"
          >
            $
            123.00
          </div>
          <p>
            item descriotion
          </p>
          <div
            class="right"
          >
            QTY: 
            3
          </div>
        </div>
        <div
          class="card-action"
        >
          <div
            class="item-rating-container left"
          >
            <a
              class="rating-star-link"
              href="/"
            >
              <i
                class="material-icons tiny"
              >
                star
              </i>
            </a>
            <a
              class="rating-star-link"
              href="/"
            >
              <i
                class="material-icons tiny"
              >
                star
              </i>
            </a>
            <a
              class="rating-star-link"
              href="/"
            >
              <i
                class="material-icons tiny"
              >
                star
              </i>
            </a>
            <a
              class="rating-star-link"
              href="/"
            >
              <i
                class="material-icons tiny"
              >
                star
              </i>
            </a>
            <a
              class="rating-star-link"
              href="/"
            >
              <i
                class="material-icons tiny"
              >
                star_border
              </i>
            </a>
          </div>
          <button
            class="waves-effect waves-light btn btn-small red right"
          >
            X
          </button>
          <button
            class="waves-effect waves-light btn btn-small orange right"
          >
            Update
          </button>
          <a
            class="right"
            href="/items/undefined"
          >
            View Details
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
